### PR TITLE
Validate declarations format

### PR DESCRIPTION
**Pull request by Ndpnt**
Improvement of #841 
Improve clarity of error message:

Capture d’écran 2022-05-10 à 11 42 04

Also add a bin script to allow fixing declarations both from OTA and declarations repository without having `eslint` in `dependencies`, only `devDependencies`.
From OTA there is the command `npm run declarations:fix` and the `ota-fix-declarations` script is exposed in the same way as `ota-validate`. In the declarations repository we can have the same npm command bby adding in the `package.json`: 
```
"scripts": {
    "test": "ota-validate",
    "declarations:fix": "ota-fix-declarations"
},
```

So all declarations repositories should be updated if this PR is merged to add npm command.

We should also add doc for available commands in the declarations repositories README. See https://github.com/OpenTermsArchive/template-declarations/issues/6

PS: I did not add a suggestion in the tests report because when there is a `:` in the error message the output becomes ugly…:
Capture d’écran 2022-05-10 à 11 52 40

### CODE REVIEW

**Comment by Ndpnt** [comment]
Maybe we should rename `validate` commands for consistency.
From this:
```
"scripts": {
    "dataset:generate": "node scripts/dataset/main.js",
    "dataset:release": "node scripts/dataset/main.js --publish --remove-local-copy",
    "dataset:scheduler": "npm run dataset:release -- --schedule",
    "declarations:fix": "node ./bin/fix-declarations.js",
    "lint": "eslint src test scripts bin",
    "lint:fix": "npm run lint -- --fix",
    "refilter": "node src/main.js --refilter-only",
    "start": "node --max-http-header-size=32768 src/main.js",
    "start:scheduler": "node --max-http-header-size=32768 src/main.js --schedule",
    "test": "cross-env NODE_ENV=test mocha --recursive \"./src/**/*.test.js\" \"./scripts/**/*.test.js\" --exit",
    "posttest": "npm run lint",
    "test:debug": "npm run test -- --inspect-brk --exit",
    "validate": "node ./bin/validate.js",
    "validate:schema": "npm run validate -- --schema-only"
},
```

to this: 
```
"scripts": {
    "dataset:generate": "node scripts/dataset/main.js",
    "dataset:release": "node scripts/dataset/main.js --publish --remove-local-copy",
    "dataset:scheduler": "npm run dataset:release -- --schedule",
    "declarations:fix": "node ./bin/fix-declarations.js",
    "declarations:validate": "node ./bin/validate.js",
    "declarations:validate:schema": "npm run validate -- --schema-only",
    "lint": "eslint src test scripts bin",
    "lint:fix": "npm run lint -- --fix",
    "refilter": "node src/main.js --refilter-only",
    "start": "node --max-http-header-size=32768 src/main.js",
    "start:scheduler": "node --max-http-header-size=32768 src/main.js --schedule",
    "test": "cross-env NODE_ENV=test mocha --recursive \"./src/**/*.test.js\" \"./scripts/**/*.test.js\" --exit",
    "posttest": "npm run lint",
    "test:debug": "npm run test -- --inspect-brk --exit"
},
```

What do you think?

**Review by martinratinaud — APPROVED** [review]
Reviewed and added 
- explanation within the fix-declarations file
- logs for better understanding of what happens when fixing declarations with the CLI

**Comment by martinratinaud** [comment]
@Ndpnt I propose also to harmonize `bin` actions

As actiosn are within the OTA core project, we do not know what `ota-validate` means (or at least its is not obvious)

I suggest we use
- `ota-validate-declarations`
- `ota-lint-declarations`

I prefer `lint` over `fix` as `fix` could suggest that it will update the selector and/or url and "fix" the whole document, while "lint", despite its technical connotation, is the real word for what it's exactly doing.

WDYT

**Comment by Ndpnt** [comment]
> @Ndpnt I propose also to harmonize `bin` actions
> 
> As actiosn are within the OTA core project, we do not know what `ota-validate` means (or at least its is not obvious)
> 
> I suggest we use
> 
> * `ota-validate-declarations` or `ota-test-declarations` which would be more coherent
> * `ota-lint-declarations`
> 
> I prefer `lint` over `fix` as `fix` could suggest that it will update the selector and/or url and "fix" the whole document, while "lint", despite its technical connotation, is the real word for what it's exactly doing.
> 
> WDYT

I agree with your suggestions! 👍 
Go with: `ota-validate-declarations` and `ota-lint-declarations`.

**Comment by Ndpnt** [comment]
Can you also modify npm scripts like this:
 
```
    "declarations:fix": "node ./bin/fix-declarations.js",
    "validate": "node ./bin/validate.js",
    "validate:schema": "npm run validate -- --schema-only"
```
to
```
    "declarations:fix": "node ./bin/fix-declarations.js",
    "declarations:validate": "node ./bin/validate.js",
    "declarations:validate:schema": "npm run validate -- --schema-only",
 ```

File: scripts/validation/validate.js [hunk 1]
```diff
@@ -238,3 +234,18 @@ function assertValid(schema, subject) {
     throw new Error(errorMessage);
   }
 }
+
+async function lintFile(filePath) {
+  const [lintResult] = await new ESLint({ overrideConfigFile: ESLINT_CONFIG_PATH, fix: false }).lintFiles(filePath);
```
[the remark this answers is not in the document]
Comment (Ndpnt) [reply scripts/validation/validate.js:239]: Totally right!
I forgot to do that when I factorized code.

**Pull request merged**